src/mainboard/protectli/vault_adl_n/variants/vp2430/gpio.c

```diff
@@ -0,0 +1,1415 @@
+	_PAD_CFG_STRUCT(GPP_I12, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* GPP_I13 - EMMC_DATA5 */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_I13, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_I13, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* GPP_I14 - EMMC_DATA6 */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_I14, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_I14, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* GPP_I15 - EMMC_DATA7 */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_I15, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_I15, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* GPP_I16 - EMMC_RCLK */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_I16, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_I16, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* GPP_I17 - EMMC_CLK */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_I17, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_I17, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* GPP_I18 - EMMC_RESET# */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_I18, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_I18, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* GPP_I19 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_I19, NONE),
+
+	/* ------- GPIO Group GPP_H ------- */
+
+	/* GPP_H0 - GPIO */
+	/* DW0: 0x44000200, DW1: 0x00000000 */
+	PAD_CFG_GPO(GPP_H0, 0, DEEP),
+
+	/* GPP_H1 - GPIO */
+	/* DW0: 0x44000200, DW1: 0x00000000 */
+	PAD_CFG_GPO(GPP_H1, 0, DEEP),
+
+	/* GPP_H2 - GPIO */
+	/* DW0: 0x44000200, DW1: 0x00000000 */
+	PAD_CFG_GPO(GPP_H2, 0, DEEP),
+
+	/* GPP_H3 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_H3, NONE),
+
+	/* GPP_H4 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_H4, NONE),
+
+	/* GPP_H5 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_H5, NONE),
+
+	/* GPP_H6 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_H6, NONE),
+
+	/* GPP_H7 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_H7, NONE),
+
+	/* GPP_H8 - I2C4_SDA */
+	PAD_CFG_NF(GPP_H8, NONE, DEEP, NF1),
+
+	/* GPP_H9 - I2C4_SCL */
+	PAD_CFG_NF(GPP_H9, NONE, DEEP, NF1),
+
+	/* GPP_H8 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_H8, NONE),
+
+	/* GPP_H9 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_H9, NONE),
```

Comment:
Duplicated H8 and H9. Copy-pasta error? Remove

Reply:
[Removed](https://github.com/Dasharo/coreboot/pull/576/commits/68fe32358db1c4826321169730cb09ec89ec4a40#diff-2f9e6fbf9293a4dbeecc2314eb0b449c4eb64691bfc1c58981fc506a602441bcL523)